project/backend/api/views.py

```diff
@@ -19,6 +19,20 @@
 
 # Create your views here.
 
+def get_user_role_from_request(self, request):
+    user = User.objects.get(id=request.user.id)
+    reviewer = Reviewer.objects.filter(user=user)
+    cont = Contributor.objects.filter(user=user)
+    basic = BasicUser.objects.filter(user=user)
+    if reviewer.count() != 0:
+        return "REVIEWER"
+    elif cont.count() != 0:
+        return "CONTRIBUTOR"
+    elif basic.count() != 0:
+        return "BASICUSER"
+    else:
+        return None
```

Comment:
get_user_role_from_request is useful for the implementations with the methods. however we need some additional mechanisms for the apis implemented with django rest framework. for example, in ChangeProfileSettingsView class below, we use a list called "permissions" for checking the permissions. this is a proper and easy way for such implementations. checking the authorization in the methods can be hard and may require additional implementations for such api implementations. for example, some of them doesn't even implemented with the methods and there is no other way of authorization checks. can you add some additional mechanisms for that purpose?

I have implemented one in my last pr and you can check it by [clicking here](https://github.com/bounswe/bounswe2023group9/pull/501/files#diff-65898ee62949268f02cd0b0142da9500a5f9468de99ee79dc5c6df0e44e11c2e)

It is really simple and easy to implement. btw, you can also call this method in the classes' has_permission method too.